### PR TITLE
feat: added custom commands

### PR DESCRIPTION
**Pull request by kyranet**
### Pending to do:

- [x] Default to single-part Sentence if parse fails.
- [x] Handling invalid arguments correctly.
- [x] Add `mockcase`, ~~`id`~~, `length`, and ~~`name`~~ transformers.

### CODE REVIEW

File: src/lib/database/entities/GuildEntity.ts [hunk 1]
```diff
@@ -35,7 +36,7 @@ export class GuildEntity extends BaseEntity {
 	public disabledCommands: string[] = [];
 
 	@ConfigurableKey({ dashboardOnly: true, description: LanguageKeys.Settings.DashboardOnlyKey, type: 'customcommand', array: true })
-	@Column('jsonb', { name: 'custom-commands', default: () => "'[]'::JSONB" })
+	@Column('jsonb', { name: 'custom-commands', transformer: kTagsTransformer, default: () => "'[]'::JSONB" })
```
Comment (favna) [inline src/lib/database/entities/GuildEntity.ts:39]: Does this need a migration?

File: src/languages/en-US/serializers.json [hunk 2]
```diff
@@ -4,6 +4,8 @@
 	"customCommandInvalidContent": "The property \"content\" must be a string.",
 	"customCommandInvalidEmbed": "The property \"embed\" must be a boolean.",
 	"customCommandInvalidId": "The property \"id\" must be a string.",
+	"customCommandInvalidType": "The type {{tag.type}} is not valid, it must be the following: {{possibles, orList}}",
+	"customCommandInvalidType_plural": "The type {{tag.type}} is not valid, it must be one of the following: {{possibles, orList}}",
```
Comment (favna) [inline src/languages/en-US/serializers.json:8]: Read https://www.i18next.com/translation-function/plurals then fix it

File: src/lib/i18n/languageKeys/keys/CustomCommandSerializer/All.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
-import { T } from '#lib/types';
+import { FT, T } from '#lib/types';
 
 export const InvalidId = T<string>('serializers:customCommandInvalidId');
 export const InvalidEmbed = T<string>('serializers:customCommandInvalidEmbed');
 export const InvalidColor = T<string>('serializers:customCommandInvalidColor');
 export const InvalidContent = T<string>('serializers:customCommandInvalidContent');
 export const InvalidArgs = T<string>('serializers:customCommandInvalidArgs');
+export const InvalidType = FT<{ possibles: readonly string[] }, string>('serializers:customCommandInvalidType');
```
Comment (favna) [inline src/lib/i18n/languageKeys/keys/CustomCommandSerializer/All.ts]: Read https://www.i18next.com/translation-function/plurals then fix it

**Review by favna — CHANGES_REQUESTED** [review]
<snip/>

File: src/lib/database/entities/GuildEntity.ts (hunk 1 above)
Comment (kyranet) [reply src/lib/database/entities/GuildEntity.ts:39]: No, it's the same information except `args` have been removed.

**Review by favna — APPROVED** [review]

**Pull request merged**